### PR TITLE
feat: Add project and environment scoping to API keys

### PR DESCRIPTION
**Pull request by ivarconr**
Some help on improving the secrets column would be appreciated @FredrikOseberg

### CODE REVIEW

File: src/component/api-token/ApiTokenCreate/ApiTokenCreate.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        const value = event.target.value;
+        setData({...data, project: value})
+    }
+
+    const setEnvironment = (event: React.ChangeEvent<{value: string }>) => {
+        const value = event.target.value;
+        setData({...data, environment: value})
+    }
+
+    const selectableProjects = [{id: '*', name: 'ALL'}, ...(projects || [])].map(i => ({
```
Comment (FredrikOseberg) [inline src/component/api-token/ApiTokenCreate/ApiTokenCreate.tsx:109]: Are these necessary? Both hooks return an empty array if there is no data.

File: src/hooks/api/getters/useApiTokens/useApiTokens.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const useApiTokens = () => {
+    const fetcher = async () => {
+        const path = formatApiPath(`api/admin/api-tokens`);
+        const res = await fetch(path, {
+            method: 'GET',
+        });
+        return res.json();
+    };
+
+    const { data, error } = useSWR(`api/admin/api-tokens`, fetcher);
```
Comment (FredrikOseberg) [inline src/hooks/api/getters/useApiTokens/useApiTokens.ts:14]: Just a minor detail, but extracting the key into a API_TOKENS_CACHE_KEY constant makes it less error prone, and allows us to export the key.

**Review by FredrikOseberg — APPROVED** [review]
Looks excellent. I only have minor comments.

File: src/component/api-token/ApiTokenCreate/ApiTokenCreate.tsx (hunk 1 above)
Comment (ivarconr) [reply src/component/api-token/ApiTokenCreate/ApiTokenCreate.tsx:109]: A of course, did not think about what the hook actually returned. 

They will also slightly change now that you cannot select  "Any"  environment.

File: src/hooks/api/getters/useApiTokens/useApiTokens.ts (hunk 2 above)
Comment (ivarconr) [reply src/hooks/api/getters/useApiTokens/useApiTokens.ts:14]: thanks 👍🏼

File: src/component/api-token/ApiTokenList/ApiTokenList.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        }
+        setDeleteToken(undefined);
+        setShowDelete(false);
+        refetch();
+        setToastData({
+            type: 'success',
+            show: true,
+            text: 'Successfully deleted API token.',
+        });
+        setTimeout(hideToast, 2000);
```
[the remark this answers is not in the document]
Comment (ivarconr) [reply src/component/api-token/ApiTokenList/ApiTokenList.tsx:90]: I was not aware!

**Pull request merged**